lms/templates/navigation.html

```diff
@@ -21,6 +20,26 @@
 ## Provide a hook for themes to inject branding on top.
 <%block name="navigation_top" />
 
+## Add UI Toolkit components if using the Pattern Library
+% if uses_pattern_library:
+    <%block name="js_extra">
+    <%static:require_module module_name="js/header_factory" class_name="HeaderFactory">
+    HeaderFactory();
+    </%static:require_module>
+    </%block>
+% endif
+
+<%
+if uses_pattern_library:
```

Comment:
My motivation here is that I believe that we should try to contain the management of legacy code as much as possible in order to maximize maintainability of this header. Developers should be able to change the header with minimal knowledge of which stylesheet is getting applied. At the very least, I think it makes sense to move all the if/else-ing and CSS class variables definitions to the top of the file, rather than have them sprinkled inline throughout the HTML. At that point, assuming mako can do this, it might also help to move all if/else-ing to a separate file which gets imported by this template.